<?xml version="1.0"?>
<launch>

  <arg name="robot" default="$(optenv ROBOT !!ROBOT NOT SET!!)" />
  <arg name="robot_env" default="$(optenv ROBOT_ENV !!ROBOT_ENV NOT SET!!)" />

  <!-- common navigation launch files -->
  <include file="$(find mir_2dnav)/ros/launch/nav_common.launch">
    <arg name="robot_env" value="$(arg robot_env)" />
  </include>

  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <rosparam file="$(find mir_2dnav)/ros/config/trajectory/move_base_params.yaml" command="load" />
    <rosparam file="$(find mir_hardware_config)/$(arg robot)/config/base_footprint.yaml" command="load" />

    <rosparam file="$(find mir_2dnav)/ros/config/trajectory/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find mir_hardware_config)/$(arg robot)/config/base_footprint.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find mir_2dnav)/ros/config/trajectory/global_costmap_params.yaml" command="load" />

    <rosparam file="$(find mir_2dnav)/ros/config/trajectory/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find mir_hardware_config)/$(arg robot)/config/base_footprint.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find mir_2dnav)/ros/config/trajectory/local_costmap_params.yaml" command="load" />

    <rosparam file="$(find mir_2dnav)/ros/config/trajectory/base_local_planner_params.yaml" command="load" ns="TrajectoryPlannerROS" />

    <!-- Remap for twist mux -->
    <remap from="cmd_vel" to="/cmd_vel_prio_medium"/>
    <remap from="~force_field_recovery/cmd_vel" to="/cmd_vel_prio_medium"/>
  </node>

</launch>
